Applications/Calendar/CalendarWidget.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "Calendar.h"
+#include <LibGUI/Frame.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/Widget.h>
+
+class CalendarWidget final : public GUI::Widget {
+    C_OBJECT(CalendarWidget)
+
+public:
+    CalendarWidget();
+    virtual ~CalendarWidget() override;
+
+private:
+    void update_calendar_tiles(int target_month, int target_year);
+    void show_add_event_window(Calendar* calendar);
+
+    //TODO: Change to RefPtr or NonNullRefPtr...?
+    Calendar* m_calendar;
+    RefPtr<GUI::Label> m_selected_date_label;
+    RefPtr<GUI::Button> m_prev_month_button;
+    RefPtr<GUI::Button> m_next_month_button;
+
+    class CalendarTile final : public GUI::Frame {
+        C_OBJECT(CalendarTile)
+    public:
+        CalendarTile(Calendar* calendar, int index, int year, int month, int day);
+        void update_values(Calendar* calendar, int index, int year, int month, int day);
```

Comment:
If it's not valid to pass `calendar=nullptr`, these should take a `Calendar&` instead.